CKSIncrementalStore/CKSIncrementalStoreSyncOperation.swift

```diff
@@ -524,7 +524,7 @@ class CKSIncrementalStoreSyncOperation: NSOperation {
                     return true
                 })
                 var values = ckRecord.dictionaryWithValuesForKeys(keys)
-                managedObject.setValuesForKeysWithDictionary(values)
+                managedObject.setValuesForKeysWithDictionary(values) // TODO: it would be safer to filter the keys first to ensure they exist in managedObject. Over time schema changes could result in fields that only on the server - this was covered in one of the videos from WWDC 2014.
```

Comment:
Keys are already used belonging to the properties of entity to which the instance of NSManagedObjectContext belongs. Values belonging to these keys only are retrieved from the record. So in case of any new keys added on the CloudKit servers would require require to setup a migration for the local store :)